x/market/types/deal.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/x/market/util/address"
+	"strings"
+)
+
+func NewDealAddress(dealId uint64) sdk.AccAddress {
+	key := append([]byte("deal"), sdk.Uint64ToBigEndian(dealId)...)
+	return address.Module(ModuleName, key)
+}
+
+func AccDealAddressFromBech32(address string) (sdk.AccAddress, error) {
+	if len(strings.TrimSpace(address)) == 0 {
+		return sdk.AccAddress{}, fmt.Errorf("empty address string is not allowd")
+	}
+
+	bech32PrefixAccAddr := sdk.GetConfig().GetBech32AccountAddrPrefix()
+
+	bz, err := sdk.GetFromBech32(address, bech32PrefixAccAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = VerifyDealAddressFormat(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	return sdk.AccAddress(bz), nil
+}
+
+func VerifyDealAddressFormat(bz []byte) error {
+	verifier := sdk.GetConfig().GetAddressVerifier()
+	if verifier != nil {
+		return nil
+	}
+
+	if len(bz) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrUnknownAddress, "address cannot be empty")
+	}
+
+	if len(bz) > address.MaxAddrLen {
+		return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address max length is %d, got %d", address.MaxAddrLen, len(bz))
+	}
+
+	return nil
+}
```

Comment:
The Cosmos SDK v0.42 ```AccAddressFromBech32``` is different from Comsos SDK 0.43 ```AccAddressFromBech32```. The max length of address which is constant variable ```AddrLen``` in v0.42 is 20, but the max length of address in v0.43 is 255. When ```dealAddress``` use v0.42 ```AccAddressFromBech32```, it makes error. So I need to use v0.43 ```AccAddressFromBech32``` for make ```dealAddress```. I'll write comment on this function.

Reply:
I see. The dealAddress is generated by hash function, which is implemented in v0.43.x, with its dealId.
The resulting address may exceed 20 bytes. 
Is this the way osmosis did for create pool?

Reply:
Yes, Osmosis also use this function for making pool address.

Reply:
I just understood why this logic needs to be copied from cosmos-sdk. As we discussed, this workaround is what Osmosis did: https://github.com/osmosis-labs/osmosis/pull/169. Thanks for digging into this issue.

Please mark these functions or files clearly (e.g. `v044_temp`), so that we can remove them after upgrading cosmos-sdk to v0.44.

Reply:
This review is quite important topic for create-deal feature. So I will not resolve this conversation.